datasets/nasa-nex-gddp-cmip6/nasa_nex_gddp_cmip6.py

```diff
@@ -0,0 +1,178 @@
+from typing import List, Union
+import planetary_computer
+from collections import namedtuple
+
+import io
+import pystac
+
+from pctasks.core.models.task import WaitTaskResult
+from pctasks.core.storage import StorageFactory
+from pctasks.dataset.collection import Collection
+
+from operator import itemgetter
+from string import ascii_lowercase
+import itertools
+
+from kerchunk.combine import MultiZarrToZarr
+from kerchunk.hdf import SingleHdf5ToZarr
+from kerchunk.netCDF3 import NetCDF3ToZarr
+from kerchunk.utils import rename_target
+
+import adlfs
+import xarray as xr
+import xstac
+
+class Cmip6Collection(Collection):
+    def get_item(cls, paths, range_compress=False):
+        credential = planetary_computer.sas.get_token("nasagddp", "nex-gddp-cmip6").token
+        fs = adlfs.AzureBlobFileSystem("nasagddp", credential=credential)
+        template = pystac.Item(
+            "item",
+            geometry={
+                "type": "Polygon",
+                "coordinates": [
+                    [
+                        [180.0, -90.0],
+                        [180.0, 90.0],
+                        [-180.0, 90.0],
+                        [-180.0, -90.0],
+                        [180.0, -90.0],
+                    ]
+                ],
+            },
+            bbox=[-180, -90, 180, 90],
+            datetime=None,
+            properties={"start_datetime": None, "end_datetime": None},
+        )
+
+        datasets = [xr.open_dataset(fs.open(path)) for path in paths]
+        ds = xr.merge(datasets)
+        r = xstac.xarray_to_stac(ds, template, reference_system=4326)
+        r.id = '.'.join(itemgetter(0,1,3)(cls.path_to_item_id(cls, paths[0]).split('.')))
+        for path in paths:
+            var_name = path.split("/")[6]
+            asset = pystac.Asset(
+                f"https://nasagddp.blob.core.windows.net/{path}",
+                media_type="application/netcdf",
+                roles=["data"],
+                extra_fields={"cmip6:variable": var_name}
+            )
+            r.add_asset(var_name, asset)
+        metadata = cls.create_multifile_indices(cls, paths, range_compress, ds=ds)
+        r.properties["kerchunk:indices"] = metadata
+        parts = cls.split_path(cls, paths[0])
+
+        for k, v in parts._asdict().items():
+            if 'variable' in k: continue
+            r.properties[f"cmip6:{k}"] = v
+        return r
+
+    def iter_all_strings(cls):
```

Comment:
```suggestion
    def iter_all_strings(self):
```

or `@staticmethod` if you don't want the self parameter there.

Reply:
Fixed.